File: tests/nodejs/cases/case_090_reconcile_ledger.js
/**
 * Task case 090: reconcile_ledger
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case090ReconcileLedger(input) {
  // Task: reconcile ledger records and prepare transport-ready payload.
  const taskName = 'reconcile_ledger';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ledger',
    stage: 'reconcile',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case090ReconcileLedger,
};

File: tests/python/cases/case_090_reconcile_ledger.py
"""
Task case 090: reconcile ledger payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_090_reconcile_ledger(input_data):
    # Task: reconcile ledger records and prepare transport-ready payload.
    task_name = 'reconcile_ledger'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ledger',
        'stage': 'reconcile',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']
